src/R/Support/Impl/Help/Packages/PackageIndex.cs

```diff
@@ -201,38 +200,55 @@ public sealed class PackageIndex : IPackageIndex {
             }
         }
 
+        private async Task UpdateInstalledPackagesAsync() {
+            var token = await _buildIndexLock.WaitAsync();
```

Comment:
Before code was doing `_buildIndexLock.EnqueueReset();` and then trying to rebuild the entire index. Now it only handles change in the set of packages, leaving the rest of the index intact.